cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -0,0 +1,242 @@
+			// we start with the item right after the cursor
+			start.TxIdx++
+		}
+		if request.Pagination.Limit > 0 {
+			limit = request.Pagination.Limit
+		}
+	}
+
+	// Iterate through each ledger and its transactions until limit or end range is reached
+	var txns []TransactionInfo
+	var cursor transactions.Cursor
+LedgerLoop:
+	for ledgerSeq := start.LedgerSequence; ledgerSeq <= request.EndLedger; ledgerSeq++ {
+		// Get ledger close meta from db
+		ledger, found, err := h.ledgerReader.GetLedger(ctx, ledgerSeq)
+		if (err != nil) || (!found) {
+			if err == nil {
+				err = errors.New("ledger close meta not found")
+			}
+			return GetTransactionsResponse{}, &jrpc2.Error{
+				Code:    jrpc2.InvalidParams,
+				Message: err.Error(),
+			}
+		}
+
+		// Initialise tx reader.
+		reader, err := ingest.NewLedgerTransactionReaderFromLedgerCloseMeta(h.networkPassphrase, ledger)
+		if err != nil {
+			return GetTransactionsResponse{}, &jrpc2.Error{
+				Code:    jrpc2.InvalidParams,
+				Message: err.Error(),
+			}
+		}
+
+		// Move the reader to specific tx idx
+		startTxIdx := 0
+		if ledgerSeq == start.LedgerSequence {
+			startTxIdx = int(start.TxIdx)
+		}
+		err = reader.Seek(startTxIdx)
+		if err != nil {
+			// Seek returns EOF so we can move onto next ledger
+			continue
+		}
+
+		// Decode transaction info from ledger meta
+		txCount := ledger.CountTransactions()
+		for i := startTxIdx; i < txCount; i++ {
+			cursor = transactions.NewCursor(ledger.LedgerSequence(), uint32(i), 0)
+
+			tx, err := reader.Read()
+			if err != nil {
+				return GetTransactionsResponse{}, &jrpc2.Error{
+					Code:    jrpc2.InvalidParams,
+					Message: err.Error(),
+				}
+			}
+
+			txResult, txMeta, txEnvelope, err := h.getTransactionDetails(tx)
+			if err != nil {
+				return GetTransactionsResponse{}, &jrpc2.Error{
+					Code:    jrpc2.InvalidParams,
+					Message: err.Error(),
+				}
+			}
+
+			txInfo := TransactionInfo{
+				Result:           txResult,
+				Meta:             txMeta,
+				Envelope:         txEnvelope,
+				FeeBump:          tx.Envelope.IsFeeBump(),
+				ApplicationOrder: int32(tx.Index),
+				Successful:       tx.Result.Result.Successful(),
+				LedgerSequence:   uint(ledger.LedgerSequence()),
+			}
+
+			txns = append(txns, txInfo)
+			if len(txns) >= int(limit) {
+				break LedgerLoop
+			}
+		}
+	}
+
+	latestLedgerSequence, latestLedgerCloseTime, err := h.getLatestLedgerDetails(ctx)
+	if err != nil {
+		return GetTransactionsResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: err.Error(),
+		}
+	}
+
+	return GetTransactionsResponse{
+		Transactions:               txns,
+		LatestLedger:               latestLedgerSequence,
+		LatestLedgerCloseTimestamp: latestLedgerCloseTime,
+		Pagination: &TransactionsPaginationOptions{
+			Cursor: &cursor,
+			Limit:  limit,
+		},
+	}, nil
+
+}
+
+func (h *transactionsRPCHandler) getTransactionDetails(tx ingest.LedgerTransaction) ([]byte, []byte, []byte, error) {
```

Comment:
TODO: Add docs to describe each return value, otherwise it's not obvious without reading the code